servers/src/test/java/tachyon/master/lineage/recompute/TestRecomputeExecutor.java

```diff
@@ -56,7 +58,8 @@ public void recomputeLauncherTest() throws Exception {
     FileSystemMaster fileSystemMaster = Mockito.mock(FileSystemMaster.class);
 
     RecomputeExecutor executor = new RecomputeExecutor(planner, fileSystemMaster);
-    executor.heartbeat();
+    // wait for the executor to finish running
+    executor.heartbeatWithFuture().get(1, TimeUnit.SECONDS);
```

Comment:
Thanks for the fix. What do you think of using `HeartbeatScheduler` for synchronous scheduling? You can find the example use at `HeartbeatThreadTest`